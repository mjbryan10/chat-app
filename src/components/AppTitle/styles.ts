import styled, { ThemeProps, Theme } from 'styled-components';

export const Title = styled.h1<ThemeProps<Theme>>`
  color: ${({theme}) => theme.color.text ?? 'auto'}
`;