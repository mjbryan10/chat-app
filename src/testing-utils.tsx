import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }