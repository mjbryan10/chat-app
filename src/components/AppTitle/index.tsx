import React from 'react';
import * as S from './styles';
import { selectCurrentConversation } from 'features/conversation/conversationSlice';
import { useSelector } from 'react-redux';

const AppTitle = () => {
   const currentConversation = useSelector(selectCurrentConversation);

   const generateTitle = () => {
      const defaultTitle = 'Welcome to Ch@';
      if (!currentConversation) return defaultTitle;
      const { conversation, users } = currentConversation;
      if (conversation.name) return `Ch@: ${conversation.name}`;
      if (users.length <= 2) return 'Private Ch@';
      return defaultTitle;
   };
   return (
      <S.Title>
         {generateTitle()}
      </S.Title>
   );
};

export default AppTitle;
