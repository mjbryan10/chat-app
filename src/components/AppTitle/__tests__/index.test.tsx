import React from 'react';
import { render, cleanup, RenderResult } from 'testing-utils';
import AppTitle from '..';
import { useSelector } from 'react-redux';

/**
 * Mocking useSelector to provide Redux state to tests.
 * @see https://dev.to/fredrikbergqvist/mocking-redux-useselector-hook-2ale
 */
jest.mock('react-redux', () => ({
   ...jest.requireActual<any>('react-redux'),
   useSelector: jest.fn(),
}));
describe('AppTitle Component', () => {
   afterEach(() => cleanup());
   afterAll(() => jest.clearAllMocks());
   test('should render `Welcome to Ch@` on init', () => {
      const { getByText } = render(<AppTitle />);

      expect(getByText(/welcome to ch@/i)).toBeVisible();
   });

   test('should display the conversation name if there is one in store', async () => {
      const __currentConversation = {
         conversation: { name: 'testName' },
         users: [1, 2, 3],
      };
      (useSelector as jest.Mock).mockImplementationOnce(
         () => __currentConversation
      );
      const { getByText } = render(<AppTitle />);

      expect(useSelector).toBeCalled();
      expect(getByText(/testName/i)).toBeVisible();
   });

   test('should inform is `private` chat if only 2 participants', async () => {
      const __currentConversation = { conversation: { name: null }, users: [1, 2] };
      (useSelector as jest.Mock).mockImplementationOnce(() => __currentConversation);
      const { getByText } = render(<AppTitle />);

      expect(useSelector).toBeCalled();
      expect(getByText(/private/i)).toBeVisible();
   });
});
