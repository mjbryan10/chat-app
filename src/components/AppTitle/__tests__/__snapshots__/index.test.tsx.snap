// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle Component should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="sc-AxjAm caAEPc"
  >
    Welcome to Ch@
  </h1>
</DocumentFragment>
`;
